src/gpu/intel/ocl/micro_sdpa.cl

```diff
@@ -392,6 +392,21 @@ micro_sdpa(const global KEY_DATA_T *K, const global half *Q,
         tile_hbroadcast_min(&S_tile, k_mask);
 #endif
 
+#if WITH_CAUSAL_MASK
+        /* Apply causal mask */
+        for (int jj = 0; jj < (ugemm_kq_c_type_block1 * ugemm_kq_c_type_nblock1); jj++) {
```

Comment:
`index_t` or whatever is 64-bit type...

Reply:
I can do it in another commit. we treat all indexing in this kernel as integers.